platforms/ios/src/TangramMap/TGMarker.mm

```diff
@@ -51,99 +62,148 @@ - (instancetype)initWithMapView:(TGMapViewController *)mapView
         type = TGMarkerType::none;
         self.visible = YES;
         self.drawOrder = 0;
-        self.map = mapView;
+        [self map:mapView];
     }
 
     return self;
 }
 
-- (void)setStylingString:(NSString *)styling
+- (NSError *)createNSError
+{
+    NSMutableDictionary* userInfo = [[NSMutableDictionary alloc] init];
+    // TODO: add enum for object error
+    // [userInfo setObject: forKey:@"TGMarker"];
+
+    return [NSError errorWithDomain:@"TGMarker"
+                               code:0
+                           userInfo:userInfo];
+}
+
+- (NSError *)stylingString:(NSString *)styling
 {
     _stylingString = styling;
     _stylingPath = nil;
 
-    if (!tangramInstance || !identifier) { return; }
+    if (!tangramInstance || !identifier) { return nil; }
+
+     if (!tangramInstance->markerSetStylingFromString(identifier, [styling UTF8String])) {
+        return [self createNSError];
+    }
 
-    tangramInstance->markerSetStylingFromString(identifier, [styling UTF8String]);
+    return nil;
 }
 
-- (void)setStylingPath:(NSString *)path
+- (NSError *)stylingPath:(NSString *)path
 {
     _stylingPath = path;
     _stylingString = nil;
 
-    if (!tangramInstance || !identifier) { return; }
+    if (!tangramInstance || !identifier) { return nil; }
+
+    if (!tangramInstance->markerSetStylingFromPath(identifier, [path UTF8String])) {
+        return [self createNSError];
+    }
 
-    tangramInstance->markerSetStylingFromPath(identifier, [path UTF8String]);
+    return nil;
 }
 
-- (void)setPoint:(TGGeoPoint)coordinates
+- (NSError *)point:(TGGeoPoint)coordinates
```

Comment:
Yes I first went what you are suggesting:

```objc
NSError* error;
BOOL success = [TGMarker point:coord error:&error];
if (!success) {
    // Use error
}
````
vs
```objc
NSError* error = [TGMarker point:coord];
if (error) {
   // Use error
}
```
I'm unsure what is the most common pattern here for iOS dev, so it can be changed to what you are  suggesting.